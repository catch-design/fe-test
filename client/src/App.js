import React, {Component} from 'react';
import Header from './containers/Header/Header';
import SignUp from './containers/SignUp/SignUp';
import Catalogue from './containers/Catalogue/Catalogue';
import Gallery from './containers/Gallery/Gallery';
import Footer from './containers/Footer/Footer';
import mainStyle from './main.scss';
import classes from './App.scss';

class App extends Component {

    render() {
        return (
            <div className={[mainStyle.grid__container, classes.homepage__container].join(' ')}>
                <Header title="A front-end test of sorts"/>
                <SignUp/>
                <Catalogue/>
                <Gallery/>
                <Footer/>
            </div>
        )
    }

}

export default App;