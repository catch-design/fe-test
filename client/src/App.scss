@import './main';

@media screen and (max-width: 320px){
  .homepage__container{
    display: -ms-flexbox;
    display: flex;
    -ms-flex-direction: column;
    flex-direction: column;
    margin: 0px auto;
  }
}