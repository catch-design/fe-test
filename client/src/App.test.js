import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter()});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("matches the snapshot ", () => {
    const appTree = shallow(<App />);
    expect(toJson(appTree)).toMatchSnapshot();
});