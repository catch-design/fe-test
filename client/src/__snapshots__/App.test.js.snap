// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot  1`] = `
<div
  className=" "
>
  <header
    title="A front-end test of sorts"
  />
  <SignUp />
  <Catalogue />
  <Gallery />
  <footer />
</div>
`;
